flutter/example/lib/main.dart

```diff
@@ -14,16 +15,30 @@ Future<void> main() async {
   await SentryFlutter.init(
     (options) {
       options.dsn = _exampleDsn;
-      // use breadcrumb tracking of WidgetsBindingObserver
-      // options.useFlutterBreadcrumbTracking();
-      // use breadcrumb tracking of platform Sentry SDKs
-      // options.useNativeBreadcrumbTracking();
+      if (UniversalPlatform.isAndroid) {
+        options.addEventProcessor(addAndroidDeviceId);
+      }
     },
     // Init your App.
     appRunner: () => runApp(MyApp()),
   );
 }
 
+FutureOr<SentryEvent> addAndroidDeviceId(
+  SentryEvent event, {
+  dynamic hint,
+}) async {
+  var info = await DeviceInfoPlugin().androidInfo;
+  var user = (event.user ?? SentryUser()).copyWith(id: info.fingerprint);
```

Comment:
There's a couple of candidates for Ids and I'm not quite sure which is should be user for which field.
I guess User.Id should be [fingerprint](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/fingerprint.html)
and Device.id should be [androidId](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/androidId.html).

Is [id](https://pub.dev/documentation/device_info_plus_platform_interface/latest/model_android_device_info/AndroidDeviceInfo/id.html) relevant? I guess not.

Reply:
we only care for `user.id` which is using the value of the SDK 4.x
we must set only the value read from https://developer.android.com/reference/android/provider/Settings.Secure#ANDROID_ID

ignore `device.id`